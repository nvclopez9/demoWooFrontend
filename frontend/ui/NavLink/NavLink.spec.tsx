// NavLink.test.tsx
import { render, screen } from '@woographql/testing';
import { NavLink } from '.';

describe('NavLink component', () => {
  it('renders correctly', () => {
    const testText = 'Test Link';
    const testHref = '/test';

    render(<NavLink href={testHref}>{testText}</NavLink>);

    // Check that the link is rendered with the correct text and href
    const link = screen.getByRole('link', { name: testText });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', testHref);
  });

  it('renders correctly with additional className', () => {
    const testText = 'Test Link';
    const testHref = '/test';
    const testClassName = 'test-class';

    render(<NavLink href={testHref} className={testClassName}>{testText}</NavLink>);

    // Check that the link is rendered with the correct text, href, and class
    const link = screen.getByRole('link', { name: testText });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', testHref);
    expect(link).toHaveClass(testClassName);
  });

  it('should match snapshot', () => {
    const { baseElement } = render(<NavLink href="/test">Test Link</NavLink>);
    expect(baseElement).toMatchSnapshot();
  });
});
