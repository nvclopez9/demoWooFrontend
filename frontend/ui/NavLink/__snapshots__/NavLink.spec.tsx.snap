// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink component should match snapshot 1`] = `
<body>
  <div>
    <a
      class="transition-colors group-hover:text-blue-400"
      href="/test"
    >
      Test Link
    </a>
  </div>
</body>
`;
